genesyscloud/conversations_messaging_supportedcontent/data_source_genesyscloud_conversations_messaging_supportedcontent.go

```diff
@@ -0,0 +1,114 @@
+package conversations_messaging_supportedcontent
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v133/platformclientv2"
+
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	rc "terraform-provider-genesyscloud/genesyscloud/resource_cache"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+)
+
+/*
+   The data_source_genesyscloud_conversations_messaging_supportedcontent.go contains the data source implementation
+   for the resource.
+*/
+
+// dataSourceSupportedContentRead retrieves by name the id in question
+func dataSourceSupportedContentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+
+	key := ""
+
+	if name, ok := d.GetOk("name"); ok {
+		key = name.(string)
+	}
+
+	if d.Get("name").(string) == "" {
+		return util.BuildDiagnosticError(resourceName, "no user search field specified", nil)
+	}
+
+	if dataSourceSupportedContentCache == nil {
+		dataSourceSupportedContentCache = rc.NewDataSourceCache(sdkConfig, hydrateSupportedContentCacheFn, getSupportedContentIdByName)
+	}
```

Comment:
`name` is a required field, so you can just write `key := d.Get("name").(string)` and don't do the checks. If the user does not provide a value for it, they will see an error about it being a required field, and this code will not be reached.